### DISCUSSION TITLE
Creating reusable Router Signals APIs in Angular 🗺️ - justangular.com

### DISCUSSION THREAD

**Discussion by eneajaho** [eneajaho/just-angular#14] [General]
https://justangular.com/blog/creating-reusable-router-signals-apis

**Comment by nzldvd90** [comment]
Very interesting!
I strongly suggest this library to perform bidirectional binding both for reading and writing query params:
👉 https://www.npmjs.com/package/@geckosoft/ngx-query-params

It's very useful when you are trying to read/write pagination params from URL query params. 

Might be useful to someone! 🙂

**Reply by eneajaho** [reply]
@nzldvd90 there is already a solution for that in ngxtension called linkedQueryParam: https://ngxtension.netlify.app/utilities/injectors/linked-query-param/
